queue_processors/queue_processor/queue_listener_daemon.py

```diff
@@ -52,28 +60,36 @@ def stop(self):
         Stops the Queue Processor Daemon, first making sure that
         the underlying client has finished
         """
-        self._client_handle.disconnect()
+        if self._client_handle is None:
+            print("Cannot safely disconnect _client_handle as it is "
+                  "running in original process. Messages might get lost. "
+                  "This happens when the process is manually stopped from CLI.")
+        else:
+            self._client_handle.disconnect()
+            self.safe_shutdown.set()
         super().stop()
-        self.safe_shutdown.set()
+
 
 
 def _wait_for_client(daemon):
     # Give 1 minute (60 s) to shutdown once stop has been called
-    timeout = 60 + daemon.stop_interval
-    was_safe_shutdown = daemon.safe_shutdown.wait(timeout=timeout)
+    was_safe_shutdown = daemon.safe_shutdown.wait(timeout=60)
 
     if not was_safe_shutdown:
-        logging.error(
-            "%s: Queue Client did not shutdown before timeout, so it was "
-            " killed ungracefully. This should be investigated as it could"
-            " cause messages to get lost.", datetime.now().isoformat())
+        log = logging.getLogger(__file__)
```

Comment:
Do you think it would be worth making all loggers consistently named?
In `daemon.py` you've called it `LOGGER`, whereas here it's `log`. Perhaps `LOGGER` for all instances? 🙂

Reply:
In that case `LOGGER` is a module variable, while this `log` is a local one. I think it's good to have the capital letters distinction, but I'll change it to `logger`.